website/static/js/quickProjectSearchPlugin.js

```diff
@@ -0,0 +1,510 @@
+            var numContributors = node.embeds.contributors.links.meta.total;
+            if (numContributors === 1) {
+                return self.getFamilyName(0, node);
+            }
+            else if (numContributors === 2) {
+                return self.getFamilyName(0, node) + ' and ' +
+                    self.getFamilyName(1, node);
+            }
+            else if (numContributors === 3) {
+                return self.getFamilyName(0, node) + ', ' +
+                    self.getFamilyName(1, node) + ', and ' +
+                    self.getFamilyName(2, node);
+            }
+            else {
+                return self.getFamilyName(0, node) + ', ' +
+                    self.getFamilyName(1, node) + ', ' +
+                    self.getFamilyName(2, node) + ' + ' + (numContributors - 3);
+            }
+
+        };
+
+         // Formats date for display
+        self.formatDate = function (node) {
+            return new $osf.FormattableDate(node.attributes.date_modified).local;
+        };
+
+
+        self.sortAlphabeticalAscending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.title.toUpperCase();
+                var B = b.attributes.title.toUpperCase();
+                return (A < B) ? -1 : (A > B) ? 1 : 0;
+            });
+            self.sortState('alphaAsc');
+        };
+
+        self.sortAlphabeticalDescending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.title.toUpperCase();
+                var B = b.attributes.title.toUpperCase();
+                return (A > B) ? -1 : (A < B) ? 1 : 0;
+            });
+            self.sortState('alphaDesc');
+        };
+
+        self.sortDateAscending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.date_modified;
+                var B = b.attributes.date_modified;
+                return (A < B) ? -1 : (A > B) ? 1 : 0;
+            });
+            self.sortState('dateAsc');
+        };
+
+        self.sortDateDescending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.date_modified;
+                var B = b.attributes.date_modified;
+                return (A > B) ? -1 : (A < B) ? 1 : 0;
+            });
+            self.sortState('dateDesc');
+        };
+
+        // Sorts nodes depending on current sort state.
+        self.sortBySortState = function () {
+            switch (self.sortState()) {
+                case 'alphaAsc':
+                    self.sortAlphabeticalAscending();
+                    break;
+                case 'alphaDesc':
+                    self.sortAlphabeticalDescending();
+                    break;
+                case 'dateAsc':
+                    self.sortDateAscending();
+                    break;
+                default:
+                    self.sortDateDescending();
+            }
```

Comment:
@pattisdr I noticed that @caneruguz [steered](https://github.com/caneruguz/osf.io/pull/17#discussion-diff-50315674) you from an `if` tree towards this `switch`. Another nice idiom is to use a hash (object):

```javascript
{ alphaAsc: self.sortAlphabeticalAscending
, alphaDesc: self.sortAlphabeticalDescending
, dateAsc: self.sortDateAscending
, dateDesc: self.sortDateDescending
, null: self.sortDateDescending
  }[self.sortState()]()
```

Not needed on this PR, though. :)